src/instance.ts

```diff
@@ -47,77 +54,57 @@ export default class Instance {
         this.platformVersion = splitInstance[0];
         this.cluster = splitInstance[1];
         this.id = splitInstance[2];
-
+        
         this.serviceName = options.serviceName;
         this.serviceVersion = options.serviceVersion;
-        this.tokenProvider = options.tokenProvider;
-
-        if(options.host){
-            this.host = options.host;
-        } else{
-            this.host = `${this.cluster}.${HOST_BASE}`;
-        }
-
+        
+        this.host = options.host || `${this.cluster}.${HOST_BASE}`;
+        this.logger = options.logger || new ConsoleLogger();
+        
         this.client = options.client || new BaseClient({
             encrypted: options.encrypted,
-            host: this.host
+            host: this.host,
+            logger: this.logger
         });
-        
-        if(options.logger !== undefined){
-            this.logger = options.logger;
-        }
-        else{
-            this.logger = new ConsoleLogger();
-        }
     }
+    
 
-    request(options: RequestOptions): Promise<any> {
+    request<T>(options: RequestOptions): NetworkResponse<T>{
```

Comment:
I think `NetworkResponse` could just be named `Response`. Is there anything that users of this interface need to know about the network?

Reply:
Also, why is the request method parametrised? Can the user make the library return a specific response type?

Reply:
I used `NetworkResponse` to avoid any namespace clashes - there's a `Response` already that's imported by default. I'll rename to something else.

Also, I'll refactor the requests to use an off-the-shelf cancelable promises